### DISCUSSION TITLE
Reuse validators for non-graphql methods

### DISCUSSION THREAD

**Discussion by rohit-gohri** [MichalLytek/type-graphql#1388] [Q&A]
I have some REST api routes that take the same input as some of my resolvers, and I was wondering  if there is a way I can just reuse the InputType class to validate the payloads?

**Comment by MichalLytek** [comment]
Just use `class-validator` in your api handlers.
But there's no way to check the shape or types of fields based on the input type class.

**Reply by rohit-gohri** [reply]
> Just use class-validator in your api handlers.

But I'll have to redeclare all the type, I was trying to avoid that. 

It make sense though why we can't use it, since type-graphql builds the schema and graphql-js is the one validating
